charts/kube-state/templates/kube-state-role.yaml

```
@@ -1,18 +1,34 @@
 ################################
 ## Kube State Role/ClusterRole
 #################################
-{{- $singleNamespace := .Values.global.singleNamespace }}
-{{- if .Values.global.rbacEnabled }}
+{{- $useRoles := or .Values.global.features.namespacePools.enabled .Values.global.singleNamespace }}
+{{- $useClusterRoles := and .Values.global.rbacEnabled .Values.global.clusterRoles (not $useRoles )}}
+{{- $shouldCreateResources := and .Values.global.rbacEnabled (or .Values.global.clusterRoles .Values.global.features.namespacePools.enabled) }}
+{{- $namespaces := .Values.global.features.namespacePools.namespaces.names }}
+{{- if $useClusterRoles }}
+  {{- $namespaces = list .Release.Namespace}}
+{{- else  }}
+  {{- $namespaces = append $namespaces .Release.Namespace }}
+{{- end }}
+
+{{- $rootContext := . }}
+
+{{- if $shouldCreateResources }}
```

Can we find a better name than `shouldCreateResources` ? I'm struggling to understand what this does, and this template seems to create resources even if this is false.
```suggestion
{{- if $shouldCreateResources }}
```

---

{{- $rootContext := . }} this is also unnecessary will tune this one as well